### PR TITLE
8 cannot deserialize webhookmodel

### LINKED ISSUE #8: Cannot deserialize WebhookModel
### Brief bug description

Deserializing webhook payloads via `WebhookModel` does not work with various payloads.  As an example, the following payload as the result of a publish event is fine:
```json
{
    "data": {
        "items": [
            {
                "id": "46763b84-ac6d-4347-b18a-41f8ece0badd",
                "codename": "test_publish_published_webhook",
                "language": "default",
                "type": "publish_test",
                "collection": "default"
            }
        ],
        "taxonomies": []
    },
    "message": {
        "id": "ce7bd31a-411e-4a54-af1c-7fa46bb7c1f5",
        "project_id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        "type": "content_item_variant",
        "operation": "publish",
        "api_name": "delivery_production",
        "created_timestamp": "2022-04-12T16:02:27.0593617Z",
        "webhook_url": "https://routing.azurewebsites.net/api/KontentWebhook"
    }
}
```

whereas the payload for a workflow event change does not:
```json
{
    "data": {
        "items": [
            {
                "item": {
                    "id": "66d00376-ee1d-4ed1-9a3c-a88110e65190"
                },
                "language": {
                    "id": "00000000-0000-0000-0000-000000000000"
                },
                "transition_from": {
                    "id": "fe59bae0-ead6-4145-9ea8-acb398218338"
                },
                "transition_to": {
                    "id": "02907d8d-2639-432a-9ea7-633f9e3e050e"
                }
            }
        ]
    },
    "message": {
        "id": "a94ff0a8-39cb-4975-9b65-ba7ba7305693",
        "project_id": "513fdcd0-795a-01d5-a3d4-97bb0ad9c331",
        "type": "content_item_variant",
        "operation": "change_workflow_step",
        "api_name": "content_management",
        "created_timestamp": "2022-04-12T13:39:34.4449999Z",
        "webhook_url": "https://45d3-84-67-20-182.ngrok.io/webhooks/webhooks"
    }
}
```

There are two issues with the above:

1. **language** is defined as `"language":{"id": "00000000-0000-0000-0000-000000000000"}`, but in the c# class is a string and causes a JSON exception when deserializing: *The JSON value could not be converted to System.String. Path: $.data.items[0].language | LineNumber: 0 | BytePositionInLine: 84.*
2. **transition_from** and **transition_to** are not covered in the class. (This for me is less of a problem as they are ignored)

Also of interest are that the `language` element in both of the above messages are different; the publish event uses a codename, whereas the workflow change uses a reference identifier.

### Repro steps

Replicated in LINQPad 7, using .NET 6, System.Text.Json, and Kentico.Kontent.AspNetCore 0.12.0-beta3
```c#
void Main()
{
	"publish".Dump();
	var model = System.Text.Json.JsonSerializer.Deserialize<WebhookModel>(publishJson);

	model.Dump();

	model = System.Text.Json.JsonSerializer.Deserialize<WebhookModel>(workflowJson);

	model.Dump();
}

// You can define other methods, fields, classes and namespaces here
string publishJson = "{\"data\":{\"items\":[{\"id\":\"46763b84-ac6d-4347-b18a-41f8ece0badd\",\"codename\":\"test_publish_published_webhook\",\"language\":\"default\",\"type\":\"publish_test\",\"collection\":\"default\"}],\"taxonomies\":[]},\"message\":{\"id\":\"ce7bd31a-411e-4a54-af1c-7fa46bb7c1f5\",\"project_id\":\"aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa\",\"type\":\"content_item_variant\",\"operation\":\"publish\",\"api_name\":\"delivery_production\",\"created_timestamp\":\"2022-04-12T16:02:27.0593617Z\",\"webhook_url\":\"https://routing.azurewebsites.net/api/KontentWebhook\"}}";
string workflowJson = "{\"data\":{\"items\":[{\"item\":{\"id\":\"66d00376-ee1d-4ed1-9a3c-a88110e65190\"},\"language\":{\"id\":\"00000000-0000-0000-0000-000000000000\"}}]},\"message\":{\"id\":\"a94ff0a8-39cb-4975-9b65-ba7ba7305693\",\"project_id\":\"aaaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa\",\"type\":\"content_item_variant\",\"operation\":\"change_workflow_step\",\"api_name\":\"content_management\",\"created_timestamp\":\"2022-04-12T13:39:34.4449999Z\",\"webhook_url\":\"https://45d3-84-67-20-182.ngrok.io/webhooks/webhooks\"}}";
```

### Expected behavior

Model should be consistent and should deserialize correctly

### Test environment

 - .NET 6, macos and Windows (arm64)
 - Kentico.Kontent.AspNetCore 0.12.0-beta3

### Screenshots

Add links to screenshots, if possible.

### PR DESCRIPTION
**Pull request by pokornyd**
### Motivation

Fixes #8

### Checklist

- [ ] Code follows coding conventions held in this repo
- [ ] Automated tests have been added
- [ ] Tests are passing
- [ ] Docs have been updated (if applicable)
- [ ] Temporary settings (e.g. variables used during development and testing) have been reverted to defaults

### How to test

Webhooks deserialize to their respective models. Re-serializing does not modify the webhook's payload.

### CODE REVIEW

File: Kentico.Kontent.AspNetCore/Webhooks/Models/Common.cs [hunk 1]
```diff
@@ -32,6 +18,12 @@ public class Message
         [JsonPropertyName("id")]
         public Guid Id { get; set; }
 
+        /// <summary>
+        /// Identifier of a Kentico Kontent project.
```
Comment (Simply007) [inline Kentico.Kontent.AspNetCore/Webhooks/Models/Common.cs]: For the new code, we would like to use just "Kontent".

File: Kentico.Kontent.AspNetCore/Webhooks/Models/DeliveryWebhookModel.cs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.Json.Serialization;
+using System.Threading.Tasks;
+
+namespace Kentico.Kontent.AspNetCore.Webhooks.Models
+{
+    /// <summary>
+    /// Root object of a Kontent delivery or preview-delivery API triggered webhook.
```
Comment (Simply007) [inline Kentico.Kontent.AspNetCore/Webhooks/Models/DeliveryWebhookModel.cs:11]: There might be a link to API reference (optional).

File: README.md [hunk 3]
```diff
@@ -49,7 +49,8 @@ public void ConfigureServices(IServiceCollection services)
...
 ```
-
+## Webhooks
+Package provides models for webhook deserialization: `DeliveryWebhookModel` and `ManagementWebhookModel`, to be used with (preview) delivery API and management API triggers respectively. See [Webhooks reference](https://kontent.ai/learn/reference/webhooks-reference/) in Kontent documentation.
```
Comment (Simply007) [inline README.md:53]:
```suggestion
Package provides models for webhook deserialization: `DeliveryWebhookModel` and `ManagementWebhookModel`, to be used with (preview) delivery API and management API triggers respectively. See [Webhooks reference](https://kontent.ai/learn/reference/webhooks-reference/) in Kontent documentation.

```

**Review by Simply007 — CHANGES_REQUESTED** [review]
LGTM

Check the comments - they are more or less optional.

I have also added tests to cover the happy + unhappy path according to https://docs.microsoft.com/en-us/aspnet/core/test/middleware - so just check if it make sense and adjust if you want.

**Comment by Simply007** [comment]
* add note about target framework on https://github.com/Kentico/Home/wiki/Kentico's-best-practices-for-.csproj-files#target-framework

File: Kentico.Kontent.AspNetCore/Webhooks/Models/Common.cs (hunk 1 above)
Comment (pokornyd) [reply Kentico.Kontent.AspNetCore/Webhooks/Models/Common.cs]: Updated.

File: Kentico.Kontent.AspNetCore/Webhooks/Models/DeliveryWebhookModel.cs (hunk 2 above)
Comment (pokornyd) [reply Kentico.Kontent.AspNetCore/Webhooks/Models/DeliveryWebhookModel.cs:11]: Added webhooks reference documentation link.

**Comment by pokornyd** [comment]
> * add note about target framework on https://github.com/Kentico/Home/wiki/Kentico's-best-practices-for-.csproj-files#target-framework

Note added.

File: Kentico.Kontent.AspNetCore/Webhooks/Models/ManagementWebhookModel.cs [hunk 4]
```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Text.Json.Serialization;
+
+namespace Kentico.Kontent.AspNetCore.Webhooks.Models
+{
+    /// <summary>
+    /// Root object of a Kontent management API triggered webhook.
+    /// See <see href="https://kontent.ai/learn/reference/webhooks-reference/">webhooks reference documentation</see>  for details.
```
Comment (Simply007) [inline Kentico.Kontent.AspNetCore/Webhooks/Models/ManagementWebhookModel.cs]:
```suggestion
    /// See <see href="https://kontent.ai/learn/reference/webhooks-reference/">webhooks reference documentation</see> for details.
```

**Review by Simply007 — APPROVED** [review]
Thx

**Pull request merged**